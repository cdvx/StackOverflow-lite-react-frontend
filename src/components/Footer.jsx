import React from 'react';

export default () => (
  <div>
    {/* <!-- Footer --> */}
    <footer className="page-footer relative-bottom font-small unique-color-dark">
      {/* <!-- Footer Elements --> */}
      <hr id="thisLine"/>
      <div className="container">
        {/* <!-- Grid row--> */}

        <div className="row">
          {/* <!-- Grid column --> */}
          <div className="col-md-12 py-5">
            <div className="mb-5 flex-center">
              {/* <!-- Facebook --> */}
              <a className="fb-ic">
                <i className="fa fa-facebook fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Twitter --> */}
              <a className="tw-ic">
                <i className="fa fa-twitter fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Google +--> */}
              <a className="gplus-ic">
                <i className="fa fa-google-plus fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
            </div>
          </div>
          {/* <!-- Grid column --> */}
        </div>
        {/* <!-- Grid row--> */}
      </div>
    <div className="footer-copyright text-center py-3">
      Â© 2018 Copyright:
        <a href="#">StackOverflow-lite.com</a>
      </div>
      {/* <!-- Copyright --> */}
    </footer>
  </div>
);
