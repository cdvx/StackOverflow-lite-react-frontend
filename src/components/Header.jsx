import React, { Component } from 'react';

class Header extends Component {

    logOut = event => {
        localStorage.clear();
    };

    render() {
        return (  
            <header>
                <nav  id="head-footer" className="navbar navbar-default">
                    <div className="container-fluid">
                        <div className="navbar-header">
                        <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                            <span className="sr-only">Toggle navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                        <a className="navbar-brand " id="select-question" href="/"><p>StackOverflow-lite</p></a>
                        </div>
    
                        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                        <ul className="nav navbar-nav">
                            <li className="active">{localStorage.getItem("token") && <a id="select-question" href="/postquestion">Post Question <span className="sr-only">(current)</span></a>}</li>
                            
                        </ul>
                        <ul className="nav navbar-nav navbar-right">
                            {!localStorage.getItem("token") && <li className="active"><a id="select-question" href="/login">Login</a></li>}
                            {!localStorage.getItem("token") && <li className="active"><a id="select-question" href="/signup">SignUp</a></li>}
                            {localStorage.getItem("token") && <li className="active"><a id="select-question" className="logOut" href="/" onClick={this.logOut} >Sign out</a></li>}
                        </ul>
                        </div>
                    </div>
                </nav>
            </header>
        );
    }
}

 
export default Header;