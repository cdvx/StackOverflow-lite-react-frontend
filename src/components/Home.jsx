import React from 'react';
import Header from "./Header";
import Footer from "./Footer";
import SignUp from "./SignUp";
import "../assets/styles.css";

class Home extends React.Component {
  state = {
  };
  render() {
    return (
      <React.Fragment>
        <div className="jumbotron2">
        </div>
      </React.Fragment>
    );
  }
}

export default Home;
