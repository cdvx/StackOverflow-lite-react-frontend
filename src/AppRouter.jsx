import React from "react";
import { Route, BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import Root from "./Root";
import Home from "./components/Home";
import signUp from "./components/SignUp";
import store from "./store";

const AppRouter = () => (
    <Provider store={store}>
      <Router>
        <React.Fragment>
          <Root>
            <div>
              <Route exact path="/" component={Home} />
              {/* <Route path="/login" component={Login} /> */}
              <Route exact path="/signup" component={signUp} />
              <Route path="/home" component={Home} />
            </div>
          </Root>
        </React.Fragment>
      </Router>
    </Provider>
  );
  export default AppRouter;