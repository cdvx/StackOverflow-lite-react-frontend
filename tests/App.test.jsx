import React from 'react';
import {shallow,  mount } from 'enzyme';
import AppRouter from '../src/AppRouter';
import Home from "../src/components/Home";
import Header from "../src/components/Header";
import Footer from "../src/components/Footer";


describe('<AppRouter />', () => {
  it('should render without crashing', () => {
    expect(mount.bind(null, <AppRouter />)).toMatchSnapshot();
  });
});


describe("<Home />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<Header />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".active").first().simulate("click"));
    expect(wrapper.find(".active").last().simulate("click"));
  });
});


describe("<Footer />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});