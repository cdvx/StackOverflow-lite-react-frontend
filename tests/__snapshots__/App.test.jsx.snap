// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> should render without crashing 1`] = `[Function]`;
